### DISCUSSION TITLE
site to site ipsec tunnel: telnet to host behind ipsec host not working

### DISCUSSION THREAD

**Discussion by nathanatimu** [strongswan/strongswan#1419] [Q&A] [answered]
Hi All,
I'm having an issue and was hoping you can help me out in this.

OS = ubuntu 22.04
Host Y: Linux strongSwan U5.9.5/K5.15.0-52-generi
Host Z: Linux strongSwan U5.9.5/K5.15.0-1022-aws

host A = 192.168.1.110
host Y = 192.168.1.70
host Z = 10.10.0.10

i've set up a site to site ipsec tunnel between a on prem host (lets call this host Y) and a host in AWS. (lets call this host Z)
The tunnel is up and running.

I can open a telnet session from Host Y to Z and vice versa on port 22. so far so good.

When i try to open a telnet session from a host (lets call this one host A) that sits behind host Y to host Z, the telnet connection is not being established. Vice versa, trying telnet from host Z to a host behind host Y wont work either. 
However a ping does work in both scenario's...
Also a telnet from host Y to host A is working.

the route on the host behind Y is:
`<subnet-behind-host-Z> via <ip-of-host-Y> dev ens8`

the route table of host Z is configured via an AWS route table and routes the subnet behind host Y via host Z.

Also all hosts have packet forwarding configured:
`net.ipv4.ip_forward = 1
net.ipv6.conf.all.forwarding = 1
net.ipv4.conf.all.accept_redirects = 0
net.ipv6.conf.all.accept_redirects = 0`

i'm a bit lost if this is routing issue of or a ipsec issue... 

so here comes the config of ipsec.conf:
host Y:

config setup
        charondebug="all"
        uniqueids=yes
conn bla-to-bla
        type=tunnel
        auto=start
        keyexchange=ikev2
        authby=secret
        left=HOST Y
        leftsubnet=192.168.0.0/16
        right=Host Z
        rightsubnet=10.10.0.0/16
        ike=aes256-sha1-modp1024!
        esp=aes256-sha1!
        aggressive=no
        keyingtries=%forever
        ikelifetime=28800s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart

host Z:
config setup
        charondebug="all"
        uniqueids=yes
conn bla-to-bla
        type=tunnel
        auto=start
        keyexchange=ikev2
        authby=secret
        left= Host Z
        leftsubnet=10.10.0.0/16
        leftid=EIP-of-Host Z
        right=HOST Y
        rightsubnet=192.168.0.0/16
        ike=aes256-sha1-modp1024!
        esp=aes256-sha1!
        aggressive=no
        keyingtries=%forever
        ikelifetime=28800s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart

Please let me know in case of questions
Thanks for any help in this!

**update**
when i do a tcpdump on host Y (for traffic from host A to host Z), i see the packages coming in from host A on host Y interface ens7.

> listening on ens7, link-type EN10MB (Ethernet), snapshot length 262144 bytes
15:32:18.990933 IP host-A.40762 > host-Z.ssh: Flags [S], seq 2929193634, win 64240, options [mss 1460,sackOK,TS val 2328419630 ecr 0,nop,wscale 7], length 0
15:32:20.016366 IP host-A.40762 > host-Z.ssh: Flags [S], seq 2929193634, win 64240, options [mss 1460,sackOK,TS val 2328420655 ecr 0,nop,wscale 7], length 0
15:32:22.032393 IP host-A.40762 > host-Z.ssh: Flags [S], seq 2929193634, win 64240, options [mss 1460,sackOK,TS val 2328422671 ecr 0,nop,wscale 7], length 0
15:32:26.128392 IP host-A.40762 > host-Z.ssh: Flags [S], seq 2929193634, win 64240, options [mss 1460,sackOK,TS val 2328426767 ecr 0,nop,wscale 7], length 0
15:32:34.320428 IP host-A.40762 > host-Z.ssh: Flags [S], seq 2929193634, win 64240, options [mss 1460,sackOK,TS val 2328434959 ecr 0,nop,wscale 7], length 0
15:32:50.448445 IP host-A.40762 > host-Z.ssh: Flags [S], seq 2929193634, win 64240, options [mss 1460,sackOK,TS val 2328451087 ecr 0,nop,wscale 7], length 0
15:33:22.960626 IP host-A.40762 > host-Z.ssh: Flags [S], seq 2929193634, win 64240, options [mss 1460,sackOK,TS val 2328483599 ecr 0,nop,wscale 7], length 0

however i dont see them on host-Z when i do a tcpdump.

**update 2**
using `tcpdump -nn  host 192.168.1.110` on host Y when doing a ping from host A to host Z
the out put of the tcpdump on host Y =

used a new tcpdump command, now the reply is present.
> tcpdump -i ens7 && dst 10.10.0.10
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on ens7, link-type EN10MB (Ethernet), snapshot length 262144 bytes
19:42:46.687967 IP host-A > 10.10.0.10: ICMP echo request, id 28, seq 12, length 64
19:42:46.695672 IP 10.10.0.10 > host-A: ICMP echo reply, id 28, seq 12, length 64
19:42:47.689431 IP host-A > 10.10.0.10: ICMP echo request, id 28, seq 13, length 64
19:42:47.697159 IP 10.10.0.10 > host-A: ICMP echo reply, id 28, seq 13, length 64
19:42:48.690944 IP host-A > 10.10.0.10: ICMP echo request, id 28, seq 14, length 64
19:42:48.698523 IP 10.10.0.10 > host-A: ICMP echo reply, id 28, seq 14, length 64
19:42:49.692208 IP host-A > 10.10.0.10: ICMP echo request, id 28, seq 15, length 64
19:42:49.699812 IP 10.10.0.10 > host-A: ICMP echo reply, id 28, seq 15, length 64

when doing 
`tcpdump -s 0 -n -i nflog:5 `
or 
`tcpdump -n -s0 -p -i ens7 esp or udp port 4500`

no packages are captured when the ping is running
also when doing a telnet from host A to host Z no package are captured on host Y

**update 3**

after tweaking the tcpdump commands i can see the the following:
on host A i'm doing the following command:
`telnet 10.10.0.10 22`
and:
`sudo tcpdump -nn port 22 -i ens8`
the tcp dump results in:

> sudo tcpdump -nn port 22 -i ens8
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on ens8, link-type EN10MB (Ethernet), snapshot length 262144 bytes
20:15:01.448675 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345382091 ecr 0,nop,wscale 7], length 0
20:15:02.477166 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345383119 ecr 0,nop,wscale 7], length 0
20:15:04.493214 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345385135 ecr 0,nop,wscale 7], length 0
20:15:08.685213 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345389327 ecr 0,nop,wscale 7], length 0
20:15:16.877233 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345397519 ecr 0,nop,wscale 7], length 0
20:15:33.005180 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345413647 ecr 0,nop,wscale 7], length 0

simultanious i do on host Y:
`tcpdump -nn port 22 -i ens7`
this give this results

> tcpdump -nn port 22 -i ens7
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on ens7, link-type EN10MB (Ethernet), snapshot length 262144 bytes
20:15:33.004875 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345413647 ecr 0,nop,wscale 7], length 0
20:16:06.540395 IP 192.168.1.110.45946 > 10.10.0.10.22: Flags [S], seq 2194544238, win 64240, options [mss 1460,sackOK,TS val 2345447183 ecr 0,nop,wscale 7], length 0
20:17:38.684812 IP 192.168.1.110.57392 > 10.10.0.10.22: Flags [S], seq 234302115, win 64240, options [mss 1460,sackOK,TS val 2345539328 ecr 0,nop,wscale 7], length 0
20:17:39.692402 IP 192.168.1.110.57392 > 10.10.0.10.22: Flags [S], seq 234302115, win 64240, options [mss 1460,sackOK,TS val 2345540335 ecr 0,nop,wscale 7], length 0
20:17:41.708375 IP 192.168.1.110.57392 > 10.10.0.10.22: Flags [S], seq 234302115, win 64240, options [mss 1460,sackOK,TS val 2345542351 ecr 0,nop,wscale 7], length 0
20:17:45.868353 IP 192.168.1.110.57392 > 10.10.0.10.22: Flags [S], seq 234302115, win 64240, options [mss 1460,sackOK,TS val 2345546511 ecr 0,nop,wscale 7], length 0
20:17:54.060389 IP 192.168.1.110.57392 > 10.10.0.10.22: Flags [S], seq 234302115, win 64240, options [mss 1460,sackOK,TS val 2345554703 ecr 0,nop,wscale 7], length 0
20:18:10.188274 IP 192.168.1.110.57392 > 10.10.0.10.22: Flags [S], seq 234302115, win 64240, options [mss 1460,sackOK,TS val 2345570831 ecr 0,nop,wscale 7], length 0
20:18:44.236362 IP 192.168.1.110.57392 > 10.10.0.10.22: Flags [S], seq 234302115, win 64240, options [mss 1460,sackOK,TS val 2345604879 ecr 0,nop,wscale 7], length 0

this leads me to believe that the package are not being forwarded to from host Y to host Z.

**update 4**
added extrra UFW information:

host Y has UFW enabled, whitelisted host Z public ip for ports:
- 500
- 4500
- all traffic from 10.10.0.0/16 is allowed to connect on all ports (for testing purposes)

host Z has UFW disabled, access managed via AWS security groups, whitelisted ports for Host Y:
- 500
- 4500
- a ingresss rule for subnet 192.168.0.0/16 for all traffic. (for testing purposes)
- a egresss rule for subnet 192.168.0.0/16 for all traffic. (for testing purposes)

ip route on host Y

> default via public-ip-host-Y dev ens3
default via public-network-ip-from-dc dev ens3 proto dhcp src public-ip-host-Y metric 100
public-subnet-from-dc/24 dev ens3 proto kernel scope link src public-ip-host-Y metric 100
public-network-ip-from-dc dev ens3 proto dhcp scope link src public-ip-host-Y metric 100
192.168.0.0/16 dev ens7 proto kernel scope link src 192.168.1.70

**note. above redacted public ip's all belong to the same subnet

ip route on host Z

> 
default via 10.10.0.1 dev ens5 proto dhcp src 10.10.0.10 metric 100
10.10.0.0/20 dev ens5 proto kernel scope link src 10.10.0.10 metric 100
10.10.0.1 dev ens5 proto dhcp scope link src 10.10.0.10 metric 100
10.10.0.2 dev ens5 proto dhcp scope link src 10.10.0.10 metric 100

**update 5**
it seems that host Y is not fowarding its package correctly. during a tcp dump i can see that the package arrive on this host, but they dont arrive on the destination

**Comment by fractalb** [comment]
What are the IP addresses (that you used for testing) of HOST Y, Z and A telnet connections?

**Reply by nathanatimu** [reply]
@fractalb  they are all internal
host A = 192.168.1.110
host Y = 192.168.1.70
host Z = 10.10.0.10

**Comment by fractalb** [comment]
Did you see ESP packets leaving the host Y in your tcpdump? Did you compare the tcpdumps ( ping traffic vs telnet traffic)?
Source and destination addresses are not visible in the above tcpdump. Better use `tcpdump -nn` option

**Reply by nathanatimu** [reply]
@fractalb see update 2 in description please (thought it would be better to keep all information together)

**Reply by fractalb** [reply]
How come you see only the icmp reply packets but not the request packets? Did `ipsec statusall` or `swanctl -l` show any packet stats?

**Reply by nathanatimu** [reply]
@fractalb i updated my tcpdump command, now the reply packets are present, see also update 2. thanks!

**Reply by nathanatimu** [reply]
@fractalb i updated the description with update 3 fyi

**Comment by nathanatimu** [comment, marked as answer]
so finanlly figured it out. i need to forward packets via iptables from host Y to host Z.
so in the end below commands on host Y solved it for me:

> iptables -D ufw-user-forward -p tcp --source 192.168.1.0/24 --destination 10.10.0.0/16 -j ACCEPT
 iptables -D ufw-user-forward -p tcp --source 10.10.0.0/16 --destination 192.168.1.0/24 -j ACCEPT

**Reply by fractalb** [reply]
A proper route setup should be enough to forward the packets. You don't need these `iptables` rules. These rules only enable to route the TCP packets but not UDP or ICMP packets

**Reply by nathanatimu** [reply]
i was playing around with this, and eventually i decided to use ufw to route the traffic:

> ufw route allow in on ens7 out on ens3 to 10.10.0.0/16
ufw route allow in on ens3 out on ens7 to 192.168.0.0/16

not sure if this is what you mean
